packages/shared/src/hooks/post/usePostShareLoop.ts

```diff
@@ -0,0 +1,42 @@
+import { useState } from 'react';
+import { useActiveFeedNameContext } from '../../contexts';
+import { useFeature } from '../../components/GrowthBookProvider';
+import { feature } from '../../lib/featureManagement';
+import { useMutationSubscription } from '../mutationSubscription';
+import { upvoteMutationKey, UseVotePostMutationProps } from '../vote';
+import { Post } from '../../graphql/posts';
+
+interface UsePostShareLoop {
+  shouldShowOverlay: boolean;
+  onInteract: () => void;
+}
+
+export const usePostShareLoop = (post: Post): UsePostShareLoop => {
+  const { feedName } = useActiveFeedNameContext();
+  const [justUpvoted, setJustUpvoted] = useState(false);
+  const [hasInteracted, setHasInteracted] = useState(false);
+  const shareLoopsEnabled = useFeature(feature.shareLoops);
+  const shouldShowOverlay = justUpvoted && !hasInteracted && shareLoopsEnabled;
+
+  useMutationSubscription({
+    matcher: ({ status, mutation }) => {
+      const key = [...upvoteMutationKey, { feedName }];
```

Comment:
Should we define the `key` variable out of the matcher function? Otherwise we kind of keep redefining this every time the `matcher` function is called.

Reply:
It is not working, so I had to fix it.

Reply:
Yeah, we can introduce a memoized variable outside.